pkg/storage/tpr/install_types.go

```diff
@@ -45,20 +61,48 @@ func InstallTypes(cs clientset.Interface) error {
 		}
 
 		glog.Infof("Creating Third Party Resource Type: %s", tpr.Name)
-		if _, err := tprs.Create(&tpr); err != nil {
-			glog.Errorf("Failed to create Third Party Resource Type: %s (%s))", tpr.Name, err)
+		errChan = make(chan error, 1)
+		defer close(errChan)
+		errChans[tpr.Name] = errChan
```

Comment:
the only problem with `(sync.WaitGroup).Wait` is that it cannot time out. However, I would like to see these error channels all funnel into a single error channel, so that we can `select` on that single channel and time out as necessary. This combined with a `sync.WaitGroup` is pretty handy. For example: https://play.golang.org/p/2nn2Aqukib